node/src/components/small_network/error.rs

```diff
@@ -189,6 +189,9 @@ pub enum ConnectionError {
     /// This is usually a bug.
     #[error("handshake sink/stream could not be reunited")]
     FailedToReuniteHandshakeSinkAndStream,
+    /// Peer is running a different chainspec.
+    #[error("peer is on different chainspec")]
+    WrongChainspecHash,
```

Comment:
Let's move this variant to immediately after `IncompatibleVersion` to keep it alongside related errors please.

Also, please include the peer's chainspec hash similar to what we do for `WrongNetwork` and `IncompatibleVersion` to help with debugging.